plugins/inputs/mysql/v2/convert.go

```diff
@@ -77,6 +77,59 @@ func parseWsrepLatency(value sql.RawBytes) (interface{}, error) {
 	return result, nil
 }
 
+// ParseWsrepProviderOptions parses the given sql.RawBytes value into a map
+// containing all wsrep Provider options settings.
+func parseWsrepProviderOptions(value sql.RawBytes) (interface{}, error) {
+	parts := strings.Split(string(value), "; ")
+	result := make(map[string]interface{})
+	for _, setting := range parts {
+		key, data, found := strings.Cut(setting, " = ")
+
+		// empty string at the end of the field, skip
+		if len(setting) != 0 && !found {
+			return nil, fmt.Errorf("invalid key-value pair for %q", setting)
+		}
+
+		// Only process gcache.size for now
+		if key != "gcache.size" {
+			continue
+		}
+		key = strings.Replace(key, ".", "_", -1)
+		// Extract the size suffix for values like 128M
+		suffix := data[len(data)-1:]
```

Comment:
Please add an empty line to separate the two processing parts...

```suggestion

		// Extract the size suffix for values like 128M
		suffix := data[len(data)-1:]
```